operator/v1alpha1/types_kueue.go

```diff
@@ -0,0 +1,243 @@
+	// An empty Outputs combined with a `Replace` Strategy causes the Input resource to be ignored by Kueue.
+	// +optional
+	Outputs corev1.ResourceList `json:"outputs,omitempty"`
+}
+
+// +kubebuilder:validation:Enum=LessThanOrEqualToFinalShare;LessThanInitialShare
+type PreemptionStrategy string
+
+const (
+	LessThanOrEqualToFinalShare PreemptionStrategy = "LessThanOrEqualToFinalShare"
+	LessThanInitialShare        PreemptionStrategy = "LessThanInitialShare"
+)
+
+type FairSharing struct {
+	// enable indicates whether to enable fair sharing for all cohorts.
+	// +optional
+	Enable EnabledOrDisabled `json:"enable"`
+
+	// preemptionStrategies indicates which constraints should a preemption satisfy.
+	// The preemption algorithm will only use the next strategy in the list if the
+	// incoming workload (preemptor) doesn't fit after using the previous strategies.
+	// Possible values are:
+	// - LessThanOrEqualToFinalShare: Only preempt a workload if the share of the preemptor CQ
+	//   with the preemptor workload is less than or equal to the share of the preemptee CQ
+	//   without the workload to be preempted.
+	//   This strategy might favor preemption of smaller workloads in the preemptee CQ,
+	//   regardless of priority or start time, in an effort to keep the share of the CQ
+	//   as high as possible.
+	// - LessThanInitialShare: Only preempt a workload if the share of the preemptor CQ
+	//   with the incoming workload is strictly less than the share of the preemptee CQ.
+	//   This strategy doesn't depend on the share usage of the workload being preempted.
+	//   As a result, the strategy chooses to preempt workloads with the lowest priority and
+	//   newest start time first.
+	// The default strategy is ["LessThanOrEqualToFinalShare", "LessThanInitialShare"].
+	// +optional
+	// +kubebuilder:validation:MaxItems=2
+	PreemptionStrategies []PreemptionStrategy `json:"preemptionStrategies,omitempty"`
+}
+
+type Integrations struct {
+	// frameworks are a list of names to be enabled.
+	// Possible options:
+	//  - "batch/job"
+	//  - "kubeflow.org/mpijob"
+	//  - "ray.io/rayjob"
+	//  - "ray.io/raycluster"
+	//  - "jobset.x-k8s.io/jobset"
+	//  - "kubeflow.org/paddlejob"
+	//  - "kubeflow.org/pytorchjob"
+	//  - "kubeflow.org/tfjob"
+	//  - "kubeflow.org/xgboostjob"
+	//  - "workload.codeflare.dev/appwrapper"
+	//  - "pod"
+	//  - "deployment" (requires enabling pod integration)
+	//  - "statefulset" (requires enabling pod integration)
+	//  - "leaderworkerset.x-k8s.io/leaderworkerset" (requires enabling pod integration)
+	// +kubebuilder:validation:MaxItems=14
+	// +kubebuilder:validation:items:MaxLength=64
+	// +required
+	Frameworks []string `json:"frameworks"`
+	// externalFrameworks are a list of GroupVersionKinds
+	// that are managed for Kueue by external controllers;
+	// the expected format is `Kind.version.group.com`.
+	// As far as
+	// +optional
+	// +kubebuilder:validation:items:MaxLength=64
```

Comment:
Kubernetes groups follow the standard DNS subdomain pattern, meaning that the group itself can be up to 253 characters.